aspnetcore/fundamentals/routing.md

```diff
@@ -635,20 +809,16 @@ Values explicitly provided but that don't match a segment of the route are added
 
 ### Problems with route value invalidation
 
-As of ASP.NET Core 3.0 some URL generation schemes used in earlier ASP.NET Core versions don't work well with URL generation. The ASP.NET Core team plans to add features to address these needs in a future release. For now the best solution is to use legacy routing.
+As of ASP.NET Core 3.0, some URL generation schemes used in earlier ASP.NET Core versions don't work well with URL generation. The ASP.NET Core team plans to add features to address these needs in a future release. For now the best solution is to use legacy routing.
 
-This example demonstrates an example of a URL generation scheme that's not supported by routing.
+The following code shows an example of a URL generation scheme that's not supported by routing.
 
-```csharp
-app.UseEndpoints(endpoints =>
-{
-    endpoints.MapControllerRoute("default", "{culture}/{controller=Home}/{action=Index}/{id?}");
-    endpoints.MapControllerRoute("blog", "{culture}/{**slug}", new { controller = "Blog", action = "ReadPost", });
-});
-```
-In this example, the `culture` route parameter is used for localization. The desire is to have the `culture` parameter always accepted as an ambient value. However this doesn't work because of the way *required values* work.
-* In the first route template, the `culture` route parameter is *to the left* of `controller`, so changes to `controller` won't invalidate `culture`.
-* In the second route template, the `culture` route parameter is considered to be *to the right* of `controller`, which appears in the required values.
+[!code-csharp[](routing/samples/3.x/RoutingSample/StartupUnsupported.cs?name=snippet)]
+
+In the preceding code, the `culture` route parameter is used for localization. The desire is to have the `culture` parameter always accepted as an ambient value. However the `culture` parameter is not accepted as an ambient value because of the way required values work:
```

Comment:
```suggestion
In the preceding code, the `culture` route parameter is used for localization. The desire is to have the `culture` parameter always accepted as an ambient value. However, the `culture` parameter is not accepted as an ambient value because of the way required values work:
```